app/javascript/views/walletServices/BraveConnection.tsx

```diff
@@ -0,0 +1,98 @@
+import axios from "axios";
+import * as React from "react";
+import { FormattedMessage, injectIntl } from "react-intl";
+import routes from "../routes";
+
+import { LoaderIcon } from "brave-ui/components/icons";
+import NotConnected from "./braveConnection/NotConnected";
+import UpholdConnection from "./braveConnection/UpholdConnection";
+import GeminiConnection from "./braveConnection/GeminiConnection";
+
+// This class serves as the entry point for establishing a wallet connection.
+// It allows users to establish a connection to different crypto wallet providers.
+// Upon load makes a request to
+class BraveConnection extends React.Component<any, any> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      geminiConnection: {},
+      isLoading: true,
+      upholdConnection: {},
+    };
+  }
+
+  public componentDidMount() {
+    this.loadData();
+  }
+
+  public render() {
+    // Show the loading component first
+    if (this.state.isLoading) {
+      return <LoadingComponent />;
```

Comment:
Check out apollo graphql, when you have a chance, it has a nice way of building in this loading aspect into the first fetch

Reply:
Like it! We're using Postgres on this project but for future projects it would be awesome to use GraphQL